lighthouse-core/lib/network-request.js

```diff
@@ -297,12 +296,24 @@ module.exports = class NetworkRequest {
 
   /**
    * LR loses transfer size information, but passes it in the 'X-TotalFetchedSize' header.
+   * 'X-TotalFetchedSize' is the canonical transfer size in LR. Nothing should supersede it.
+   *
+   * The total length of the encoded data is spread out among multiple events. The sum of the
+   * values in onResponseReceived and all the onDataReceived events will equal the value
+   * seen on the onLoadingFinished event. As we process onResonseReceived and onDataReceived
+   * we accumulate the total encodedDataLength. When we process onLoadingFinished, we override
+   * the accumulated total. We do this so that if the request is aborted or fails, we still get
+   * a value via the accumulation.
+   *
+   * In Lightrider, due to instrumentation limitations, our values for encodedDataLength are bogus
+   * and not valid. However the resource's true encodedDataLength/transferSize is shared via a
+   * special response header, X-TotalFetchedSize. In this situation, we read this value from
+   * responseReceived, use it for the transferSize and ignore the encodedDataLength values in
+   * both dataReceived and loadingFinished.
    */
-  _updateTransferSizeForLightRiderIfNecessary() {
-    // Bail if we're not in LightRider, this only applies there.
-    if (!global.isLightRider) return;
+  _updateTransferSizeForLightrider() {
```

Comment:
can you also leave a comment on L70 that says go read the _updateTransferSizeForLightrider comment?